tsl/src/nodes/compressed_skip_scan/exec.c

```diff
@@ -0,0 +1,851 @@
+/*
```

Comment:
This needs a comment  describing how this node is different from the usual DecompressChunk.

I don't like that we're just copying this amount of code from some outdated version of DecompressChunk. Is there be a simpler way to achieve the alternate behavior, maybe only replacing the `.ExecCustomScan` function?